substrate/frame/broker/src/migration.rs

```diff
@@ -156,6 +162,244 @@ mod v3 {
+
+			Ok((interlude_length, configuration_leadin_length, sale_start, sale_info_leadin_length)
+				.encode())
+		}
+
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			let mut weight = T::DbWeight::get().reads(1);
+
+			if let Some(config_record) = v3::Configuration::<T>::take() {
+				log::info!(target: LOG_TARGET, "migrating Configuration record");
+
+				let updated_interlude_length: RelayBlockNumberOf<T> =
+					BlockConversion::convert_block_length_to_relay_length(
+						config_record.interlude_length,
+					);
+				let updated_leadin_length: RelayBlockNumberOf<T> =
+					BlockConversion::convert_block_length_to_relay_length(
+						config_record.leadin_length,
+					);
+
+				let updated_config_record = ConfigRecord {
+					interlude_length: updated_interlude_length,
+					leadin_length: updated_leadin_length,
+					advance_notice: config_record.advance_notice,
+					region_length: config_record.region_length,
+					ideal_bulk_proportion: config_record.ideal_bulk_proportion,
+					limit_cores_offered: config_record.limit_cores_offered,
+					renewal_bump: config_record.renewal_bump,
+					contribution_timeout: config_record.contribution_timeout,
+				};
+				Configuration::<T>::put(updated_config_record);
+			}
+			weight.saturating_accrue(T::DbWeight::get().reads_writes(1, 1));
+
+			if let Some(sale_info) = v3::SaleInfo::<T>::take() {
+				log::info!(target: LOG_TARGET, "migrating SaleInfo record");
+
+				let updated_sale_start: RelayBlockNumberOf<T> =
+					BlockConversion::convert_block_number_to_relay_height(sale_info.sale_start);
+				let updated_leadin_length: RelayBlockNumberOf<T> =
+					BlockConversion::convert_block_length_to_relay_length(sale_info.leadin_length);
+
+				let updated_sale_info = SaleInfoRecord {
+					sale_start: updated_sale_start,
+					leadin_length: updated_leadin_length,
+					end_price: sale_info.price,
+					region_begin: sale_info.region_begin,
+					region_end: sale_info.region_end,
+					ideal_cores_sold: sale_info.ideal_cores_sold,
+					cores_offered: sale_info.cores_offered,
+					first_core: sale_info.first_core,
+					sellout_price: sale_info.sellout_price,
+					cores_sold: sale_info.cores_sold,
+				};
+				SaleInfo::<T>::put(updated_sale_info);
+			}
+
+			weight.saturating_add(T::DbWeight::get().reads_writes(1, 2))
+		}
+
+		#[cfg(feature = "try-runtime")]
+		fn post_upgrade(state: Vec<u8>) -> Result<(), sp_runtime::TryRuntimeError> {
+			let (
+				old_interlude_length,
+				old_configuration_leadin_length,
+				old_sale_start,
+				old_sale_info_leadin_length,
+			): (BlockNumberFor<T>, BlockNumberFor<T>, BlockNumberFor<T>, BlockNumberFor<T>) =
+				Decode::decode(&mut &state[..]).expect("pre_upgrade provides a valid state; qed");
+
+			if let Some(config_record) = Configuration::<T>::get() {
```

Comment:
again, I don't think you need to do `if let Some()` here, and let the `else` silently fail. From what I understand, this is a `let stuff = Configuration::<T>::get().expect("must exist")`

Reply:
Configuration might not exist if pallet is not bootstrapped, in that case we shouldn't do anything I think

Reply:
sounds good, same as https://github.com/paritytech/polkadot-sdk/pull/5656#discussion_r1824160650